src/lexer/TokenizerEngine.ts

```diff
@@ -119,15 +119,16 @@ export default class TokenizerEngine {
     );
   }
 
-  // Shorthand for `match` that looks up regex from REGEX_MAP
+  // Shorthand for `match` that looks up regex from rules
   private matchToken(tokenType: TokenType): Token | undefined {
-    if (!(tokenType in this.REGEX_MAP)) {
```

Comment:
unfortunate that the `in` operator doesn't provide type narrowing with variables